File: LLD/src/main/java/org/example/p1_oops/c7_composition/composition/Engine.java
package org.example.p1_oops.c7_composition.composition;

public class Engine {
    private String type;

    public Engine(String type) {
        this.type = type;
        System.out.println("  Engine created: " + type);
    }

    public void start() {
        System.out.println("  Engine (" + type + ") started.");
    }

    public void stop() {
        System.out.println("  Engine (" + type + ") stopped.");
    }
}

File: LLD NodeJs/p1_oops/c7_composition/composition/Engine.js
/**
 * Engine class for demonstrating Composition
 * Engine is owned by Car and cannot exist independently
 */
class Engine {
    #type;

    constructor(type) {
        this.#type = type;
        console.log("  Engine created: " + type);
    }

    start() {
        console.log("  Engine (" + this.#type + ") started.");
    }

    stop() {
        console.log("  Engine (" + this.#type + ") stopped.");
    }
}

module.exports = Engine;
